### PR TITLE
rgw/notification: add opaque data

### PR DESCRIPTION
**Pull request by yuvalif**
opaque data may be set in topic configuration and later on sent inside
nottifications triggered by that topic.

Signed-off-by: Yuval Lifshitz <yuvalif@yahoo.com>

### CODE REVIEW

**Comment by yuvalif** [comment]
jenkins test make check

File: doc/radosgw/s3-notification-compatibility.rst [hunk 1]
```diff
@@ -112,10 +112,14 @@ Note that most of the API is not applicable to Ceph, and only the following acti
  - ``DeleteTopic``
  - ``ListTopics``
```
Comment (mattbenjamin) [inline doc/radosgw/s3-notification-compatibility.rst]: cool, improved

File: src/rgw/rgw_pubsub.cc [hunk 2]
```diff
@@ -562,10 +565,10 @@ int RGWUserPubSub::Bucket::remove_notification(const string& topic_name)
 }
 
 int RGWUserPubSub::create_topic(const string& name) {
-  return create_topic(name, rgw_pubsub_sub_dest(), "");
+  return create_topic(name, rgw_pubsub_sub_dest(), "", "");
```
Comment (mattbenjamin) [inline src/rgw/rgw_pubsub.cc]: could perhaps identity the empty args (e.g., "" /* opaque data */ ...

**Comment by yuvalif** [comment]
teuthology run: http://pulpito.front.sepia.ceph.com/yuvalif-2020-01-19_16:36:42-rgw:verify-wip-kefu-notif-add-opaque-data-distro-basic-smithi/

failures are similar issues to: https://github.com/ceph/ceph/pull/32618#issuecomment-575989136

only mutisite are failing: http://pulpito.front.sepia.ceph.com/yuvalif-2020-01-19_16:12:48-rgw:multisite-wip-yuval-notif-add-opaque-data-distro-basic-smithi/
(no pubsub failures)

**Comment by yuvalif** [comment]
jenkins test make check

**Pull request merged**